examples/ensemble/plot_hgbt_regression.py

```diff
@@ -0,0 +1,401 @@
+are more feature rich than -and often outperforms- alternative models like
+random forests, especially when the number of samples is larger than tens of
+thousands of samples (see
+:ref:`sphx_glr_auto_examples_ensemble_plot_forest_hist_grad_boosting_comparison.py`).
+
+The top usability features of HGBT models are:
+
+1. :ref:`categorical_support_gbdt` (see
+  :ref:`sphx_glr_auto_examples_ensemble_plot_gradient_boosting_categorical.py`).
+1. Early stopping.
+1. :ref:`nan_support_hgbt`, which avoids the need for an imputer.
+1. Support for several losses such as the :ref:`Quantile loss
+   <quantile_support_hgbdt>`.
+1. :ref:`monotonic_cst_gbdt`.
+1. :ref:`interaction_cst_hgbt`.
+
+This example aims at showcasing points 2-5 in a real life setting.
+"""
+
+# Author: Arturo Amor <david-arturo.amor-quiroz@inria.fr>
+# License: BSD 3 clause
+
+# %%
+# Preparing the data
+# ==================
+# The `electricity dataset <http://www.openml.org/d/151>`_ consists of data
+# collected from the Australian New South Wales Electricity Market. In this
+# market, prices are not fixed and are affected by supply and demand. They are
+# set every five minutes. Electricity transfers to/from the neighboring state of
+# Victoria were done to alleviate fluctuations.
+#
+# The dataset (originally named ELEC2) contains 45,312 instances dated from 7
+# May 1996 to 5 December 1998. Each sample of the dataset refers to a period of
+# 30 minutes, i.e. there are 48 instances for each time period of one day. Each
+# sample on the dataset has 5 fields: the day of week, the time stamp, the New
+# South Wales electricity demand, the Victoria electricity demand. It is
+# originally a classification task, but here we use it as a regression where the
+# target is the scheduled electricity transfer between states.
+
+from sklearn.datasets import fetch_openml
+
+electricity = fetch_openml(
+    name="electricity", version=1, as_frame=True, parser="pandas"
+)
+df = electricity.frame
+X = df.drop(columns=["transfer", "class"])
+y = df["transfer"]
+X
+
+# %%
+# This particular dataset has a stepwise constant target for the first 17,760
+# samples:
+
+y[:17760].unique()
+
+# %%
+# Let us drop those entries and explore the hourly electricity transfer over
+# different days of the week:
+
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+df = electricity.frame.iloc[17760:]
+X = df.drop(columns=["transfer", "class"])
+y = df["transfer"]
+
+fig, ax = plt.subplots(figsize=(15, 10))
+pointplot = sns.lineplot(x=df["period"], y=df["transfer"], hue=df["day"], ax=ax)
+handles, lables = ax.get_legend_handles_labels()
+ax.set(
+    title="Hourly energy transfer for different days of the week",
+    xlabel="Normalized time of the day",
+    ylabel="Normalized energy transfer",
+)
+_ = ax.legend(handles, ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"])
+
+# %%
+# Notice energy transfer increases systematically during weekends.
```

Comment:
You can also add that there is 2 peaks during the week days and mention to what it could corresponds.